### PR TITLE
Fixs #654: Add Navigation Drawer Icon Topic

### LINKED ISSUE #654: Topic should have navigation drawer icon in it
Mock: https://xd.adobe.com/view/0687f00c-695b-437a-79a6-688e7f4f7552-70b6/screen/c718fc6d-b336-475a-b31b-fd3bc84882a8/PM-EN-Play-Tab-Progress-1-

In mocks, the topic has a hamburger icon at top left corner of the screen which opens the navigation drawer.

### PR DESCRIPTION
**Pull request by PrarabdhGarg**
## Explanation

Fixes #654 

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by PrarabdhGarg** [comment]
@rt4914 I still have to make changes to the test cases. But could you just confirm if the implementation of the listener is correct

File: app/src/main/java/org/oppia/app/topic/ToolbarTitleListener.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package org.oppia.app.topic
+
+/** Interface to listen for toolbar title as they are fetched in the fragments and not activities */
+interface ToolbarTitleListener {
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/ToolbarTitleListener.kt]: Actually this is not the correct approach, the activity/fragment should be responsible for its own content and nothing else. So the title should be fetched by `TopicActivity` or `TopicFragment` only and not by any child fragment.

**Review by rt4914 — CHANGES_REQUESTED** [review]
PTAL

File: app/src/main/java/org/oppia/app/topic/ToolbarTitleListener.kt (hunk 1 above)
Comment (PrarabdhGarg) [reply app/src/main/java/org/oppia/app/topic/ToolbarTitleListener.kt]: @rt4914 I thaught that this listener would be helpful with other fragments in the future as well. Hence I wrote such a comment.
In the implementation, the responsibility of fetching the title is given to the fragment only.
Please suggest what the comment should  be if it needs to be changed

Comment (rt4914) [reply app/src/main/java/org/oppia/app/topic/ToolbarTitleListener.kt]: There are two approaches to this: 
1. Transfer toolbar to topic-fragment and set data to that toolbar from its `presenter`

2. Or, currently the fragmentpresenter is fetching the information for title, you can shift that code to `activity-presenter`.

Both are correct approach.

Comment (PrarabdhGarg) [reply app/src/main/java/org/oppia/app/topic/ToolbarTitleListener.kt]: For the second approach,  how do I retrieve the topic title in the activity presenter? I will have to copy most of the code present within the fragment presenter for that.

Comment (rt4914) [reply app/src/main/java/org/oppia/app/topic/ToolbarTitleListener.kt]: Yes, you will need to do that, meaning copy code of fragment presenter to activity presenter.

**Comment by PrarabdhGarg** [comment]
@rt4914 I have made the requested changes. PTAL

**Comment by rt4914** [comment]
Screenshot_1583750032

@PrarabdhGarg Tabs are not visible in current implementation.

**Comment by PrarabdhGarg** [comment]
@rt4914 I am unable to find the error in my present code. Could you suggest what could be a possible reason for this error?

**Comment by PrarabdhGarg** [comment]
@rt4914 I have made the changes

File: app/src/main/java/org/oppia/app/topic/TopicActivityPresenter.kt [hunk 2]
```diff
@@ -13,10 +24,19 @@ const val STORY_ID_ARGUMENT_KEY = "story_id"
...
 /** The presenter for [TopicActivity]. */
 @ActivityScope
-class TopicActivityPresenter @Inject constructor(private val activity: AppCompatActivity) {
+class TopicActivityPresenter @Inject constructor(
+  private val activity: AppCompatActivity,
+  private val logger: Logger,
+  private val topicController: TopicController
+) {
+  private var navigationDrawerFragment: NavigationDrawerFragment? = null
+  lateinit var topicId: String
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/TopicActivityPresenter.kt]: Make this `private`

File: app/src/main/java/org/oppia/app/topic/TopicActivityPresenter.kt [hunk 3]
```diff
@@ -33,6 +53,44 @@ class TopicActivityPresenter @Inject constructor(private val activity: AppCompat
...
+  private fun processTopicResult(topic: AsyncResult<Topic>): Topic {
+    if (topic.isFailure()) {
+      logger.e("TopicFragment", "Failed to retrieve topic", topic.getErrorOrNull()!!)
+    }
+    return topic.getOrDefault(Topic.getDefaultInstance())
+  }
+
+  private fun setUpNavigationDrawer() {
+    val toolbar = activity.findViewById<View>(R.id.topic_activity_toolbar) as Toolbar
+    toolbar.title = activity.getString(R.string.topic_prefix, " ")
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/TopicActivityPresenter.kt]: Remove this line. We will set text only once.

File: app/src/main/java/org/oppia/app/topic/TopicFragmentPresenter.kt [hunk 4]
```diff
@@ -28,7 +28,6 @@ class TopicFragmentPresenter @Inject constructor(
   private val topicController: TopicController
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/TopicFragmentPresenter.kt]: Remove this and logger, they are not getting used anymore.

File: app/src/main/java/org/oppia/app/topic/TopicFragmentPresenter.kt [hunk 5]
```diff
@@ -28,7 +28,6 @@ class TopicFragmentPresenter @Inject constructor(
...
 ) {
   private lateinit var tabLayout: TabLayout
-  private lateinit var topicToolbar: Toolbar
   private var internalProfileId: Int = -1
   private lateinit var topicId: String
   lateinit var storyId: String
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/TopicFragmentPresenter.kt]: Make this private

File: app/src/main/res/layout/topic_activity.xml [hunk 6]
```diff
@@ -1,7 +1,33 @@
...
-<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<androidx.drawerlayout.widget.DrawerLayout xmlns:android="http://schemas.android.com/apk/res/android"
+  xmlns:app="http://schemas.android.com/apk/res-auto"
   xmlns:tools="http://schemas.android.com/tools"
-  android:id="@+id/topic_fragment_placeholder"
+  android:id="@+id/topic_activity_drawer_layout"
   android:layout_width="match_parent"
   android:layout_height="match_parent"
-  tools:context=".topic.TopicActivity" />
+  tools:context=".home.HomeActivity">
```
Comment (rt4914) [inline app/src/main/res/layout/topic_activity.xml]: Context is incorrect here. It should be `TopicActivity`

File: app/src/sharedTest/java/org/oppia/app/topic/TopicFragmentTest.kt [hunk 7]
```diff
@@ -53,7 +53,7 @@ class TopicFragmentTest {
   @Test
   fun testTopicFragment_toolbarTitle_isDisplayedSuccessfully() {
     launchTopicActivityIntent(internalProfileId, FRACTIONS_TOPIC_ID).use {
-      onView(allOf(instanceOf(TextView::class.java), withParent(withId(R.id.topic_toolbar))))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/topic/TopicFragmentTest.kt:56]: Create a new file `TopicActivityTest` and all test cases which are related to `TopicActivity` and not `TopicFragment` shift them to that file. 

Also, add test case to check that the click on `NavigationDrawer` is working correctly.

Basically we want drawer and toolbar test cases in `TopicActivityTest` and tabs and child fragment test cases in `TopicFragmentTest` similar to how UI has been laid out.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Looks good, some nit changes and suggestion on test cases.

**Comment by rt4914** [comment]
@PrarabdhGarg If you still. busy. maybe we can close this issue so that someone else can work on this?

**Comment by PrarabdhGarg** [comment]
@rt4914 I will finish this by today

**Review by rt4914 — CHANGES_REQUESTED** [review]
If you notice correctly, the navigation drawer is not showing any profile name.

This is because of the fact that the `internalProfileId` is not getting passed correctly in `NavigationDrawerFragment`.

This can be solved by change doing following steps in `TopicActivity`
Remove `TOPIC_ACTIVITY_INTERNAL_PROFILE_ID_ARGUMENT_KEY` and replace it with `KEY_NAVIGATION_PROFILE_ID`

File: app/src/main/java/org/oppia/app/topic/TopicActivityPresenter.kt [hunk 8]
```diff
@@ -33,6 +57,43 @@ class TopicActivityPresenter @Inject constructor(private val activity: AppCompat
...
+    topicController.getTopic(profileId, topicId = topicId)
+  }
+
+  private fun getTopic(): LiveData<Topic> {
+    return Transformations.map(topicResultLiveData, ::processTopicResult)
+  }
+
+  private fun processTopicResult(topic: AsyncResult<Topic>): Topic {
+    if (topic.isFailure()) {
+      logger.e("TopicFragment", "Failed to retrieve topic", topic.getErrorOrNull()!!)
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/TopicActivityPresenter.kt]: This should be `TopicActivity`

File: app/src/sharedTest/java/org/oppia/app/topic/TopicActivityTest.kt [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.app.help.HelpActivity
+import org.oppia.app.help.HelpFragment
+import org.oppia.app.mydownloads.MyDownloadsActivity
+import org.oppia.domain.topic.FRACTIONS_TOPIC_ID
+import org.oppia.util.threading.BackgroundDispatcher
+import org.oppia.util.threading.BlockingDispatcher
+import javax.inject.Singleton
+
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/topic/TopicActivityTest.kt:31]: There are some unused imports in this file. Please remove them.

**Review by rt4914 — CHANGES_REQUESTED** [review]
PTAL

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**